workspace/teampipeline/generate_report.py

```diff
@@ -0,0 +1,203 @@
+
+
+def get_project_id(project_num):
+    query = """
+    query projectId($org_name: String!, $project_num: Int!) {
+      organization(login: $org_name) {
+        projectV2(number: $project_num) {
+          id
+          title
+        }
+      }
+    }
+    """
+    variables = {
+        "org_name": ORG_NAME,
+        "project_num": project_num,
+    }
+
+    payload = {"query": query, "variables": variables}
+    rsp = post_request(payload)
+    return rsp["data"]["organization"]["projectV2"]["id"]
+
+
+def get_project_cards(project_id):
+    query = """
+    query projectCards($project_id: ID!) {
+      node(id: $project_id) {
+        ... on ProjectV2 {
+          items(first: 100) {
+            nodes {
+              fieldValues(last: 100) {
+                nodes {
+                  ... on ProjectV2ItemFieldSingleSelectValue {
+                    name
+                    field {
+                      ... on ProjectV2FieldCommon {
+                        name
+                      }
+                    }
+                  }
+                }
+              }
+              content {
+                ... on DraftIssue {
+                  title
+                  assignees(first: 10) {
+                    nodes {
+                      login
+                    }
+                  }
+                }
+                ... on Issue {
+                  title
+                  bodyUrl
+                  assignees(first: 10) {
+                    nodes {
+                      login
+                    }
+                  }
+                }
+                ... on PullRequest {
+                  title
+                  assignees(first: 10) {
+                    nodes {
+                      login
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    """
+
+    variables = {"project_id": project_id}
+    payload = {"query": query, "variables": variables}
+    data = post_request(payload)
+
+    return sorted(
+        data["data"]["node"]["items"]["nodes"],
+        key=lambda card: card["content"]["title"],
+    )  # pragma: no cover
+
+
+def get_slack_username(github_username):
+    return {
+        "CarolineMorton": "<@UMMDD4V2P>",
+        "Jongmassey": "<@U023ZG5H24R>",
+        "StevenMaude": "<@U01TJP3CG76>",
+        "evansd": "<@UAXE5V4RG>",
+        "iaindillingham": "<@U01S6BLGK28>",
+        "inglesp": "<@U4N1YPAP7>",
+        "milanwiedemann": "<@U02GPV8NNU9>",
+        "rebkwok": "<@U01SP5JLBFD>",
+        "robinyjpark": "<@U021UP18T52>",
+        "LisaHopcroft": "<@U029ER915GX>",
+        "lucyb": "<@U035FT48KEK>",
+        "CLStables": "<@U036A6LTR7D>",
+        "tomodwyer": "<@U01UQ0T2M7V>",
+        "bloodearnest": "<@U01AMBZUT47>",
+        "ghickman": "<@U017T76R4DC>",
+        "madwort": "<@U019R5FJ7G8>",
+        "benbc": "<@U01SPCP06Q1>",
+    }.get(github_username) or f"<@{github_username}>"
```

Comment:
dict.get() throws a KeyError without a default which won't pick the rhs of the `or` here, but you can get what you want by putting that into the `default` kwarg of dict.get instead.

Reply:
`dict.get()` defaults to none if no default is specified, I thought? I have some tests that should demonstrate the behaviour, I think?

I should have been clearer with what I expected from the code review, sorry. This code was copied from the data team report. I've made some edits, like adding our names in this dict, but I've made very few other changes. I'd like to check that I've configured everything correctly and see it working. Then move it into a shared report for both teams. I want to take small steps because there are very few tests around this code and I don't want to break things for the data team, who did the majority of the work here anyway.

Reply:
> defaults to none if no default is specified

Yes, I just can't read the docs apparently 🤦‍♂️  Ignore my whole comment!